### PR TITLE
Show Mautic admin page titles

### PR DESCRIPTION
**Pull request by Yame-**
| Q | A |
| --- | --- |
| Enhancement | Yes |
| Issues addressed (#s or URLs) | #1142 |
#### Description:

Changes the title tag so that it's more informative instead of only "Mautic".
It will now read "Mautic - Email - [name_of_email]" or "Mautic - Campaigns"
#### Steps to test this PR:
1. Install this PR
2. Click on some pages and see of the title changes correctly

### CODE REVIEW

**Comment by Yame-** [comment]
Hi @jordan8037310, 
I've read that you were going to deliver a fix as well. 
Am I going in the right direction with this implementation?
Any advice would be awesome! 👍

**Comment by jordan8037310** [comment]
@Yame- this looks good! I actually started on the other side of this problem and hadn't worked on the Javascript side yet. 

I would argue that a better order is `headerTitle - Mautic` instead of vice versa. Reasoning being if you have a lot of tabs open they all shorten and then you can only see `Mautic - `

**Comment by Yame-** [comment]
Makes more sense @jordan8037310!

File: app/bundles/CoreBundle/Assets/js/1.core.js [hunk 1]
```diff
@@ -1173,6 +1174,23 @@ var Mautic = {
...
+    * Generates the title of the current page
+    *
+    * @param route
+    */
+    generatePageTitle: function(route){
+
+        if( -1 !== route.indexOf('view') ){
+            //loading view of module title
+            var currentModule = route.split('/')[3];
+            mQuery('title').html( 'Mautic - ' + currentModule[0].toUpperCase() + currentModule.slice(1) + ' - ' + mQuery('.page-header h3').html() );
```
Comment (jordan8037310) [inline app/bundles/CoreBundle/Assets/js/1.core.js]: 1186: rearrange to `pageTitle | Mautic`

File: app/bundles/CoreBundle/Assets/js/1.core.js [hunk 2]
```diff
@@ -1173,6 +1174,23 @@ var Mautic = {
...
+        } else {
+            //loading basic title
+            mQuery('title').html( 'Mautic - ' + mQuery('.page-header h3').html() );
```
Comment (jordan8037310) [inline app/bundles/CoreBundle/Assets/js/1.core.js]: Rearrange to Page Title before Mautic

File: app/bundles/CoreBundle/Views/Default/head.html.php [hunk 3]
```diff
@@ -10,7 +10,11 @@
 ?>
 <head>
     <meta charset="UTF-8" />
-    <title><?php echo $view['slots']->get('pageTitle', 'Mautic'); ?></title>
+    <title><?php echo $view['slots']->get('pageTitle', 'Mautic'); ?>
+	<?php if( !empty($view['slots']->get('headerTitle', '')) ): ?>
+		- <?php echo $view['slots']->get('headerTitle', ''); ?>
```
Comment (jordan8037310) [inline app/bundles/CoreBundle/Views/Default/head.html.php]: Rearrange to Page Title before `Mautic`. Also a pipe may be a better choice, smaller character so more real estate on tab for other characters, e.g. `Calendar | Mautic`

**Comment by Yame-** [comment]
Thanks for your suggestions @jordan8037310.
I've implemented them in my last commit.

**Comment by alanhartless** [comment]
I like this! Only issue is on some pages, the page header contains HTML (like on a contact detail's page).



Can we fix that?

**Comment by Yame-** [comment]
Thanks for the catch @alanhartless.

I've added some more code to catch those html tags and remove them.
Also, I've added a title update when an item gets saved.

**Comment by escopecz** [comment]
It looks and works good! :+1: 

Just one question, why there is "S" on every detail page title? Like `S | Contact Name Here | Mautic` or `S | Form Name | Mautic`?

**Comment by Yame-** [comment]
Hi @escopecz, 
Can't replicate your "S" problem. Detail pages seem fine on my end. 
I'm running latest version of Mautic. I've checked Chrome, Firefox and Safari. 

Any idea what could be the reason of this? 🤔

**Pull request merged**